cms/static/js/views/settings/main.js

```diff
@@ -367,12 +367,19 @@ var DetailsView = ValidatingView.extend({
             message = this.post_enrollment_email_elem.val();
         }
 
-        $.post("/settings/send_test_enrollment_email",
+        // Validation for keyword substitution
+        var validation = KeywordValidator.validate_string(message);
+        if (!validation.is_valid) {
+            message = gettext("There are invalid keywords in your email. Please check the following keywords and try again:");
+            message += "\n" + validation.invalid_keywords.join('\n');
+            alert(message);
```

Comment:
It could be, though for now I'd be content to just see it changed here (in new code).

There's also a performance consideration with implicit global variables; the interpreter has to walk the entire scope chain on each reference to resolve globals.